lib/cartopy/trace.pyx

```diff
@@ -160,73 +171,81 @@ cdef class LineAccumulator:
 cdef class Interpolator:
     cdef Point start
     cdef Point end
-    cdef projPJ src_proj
-    cdef projPJ dest_proj
+    cdef readonly transformer
     cdef double src_scale
     cdef double dest_scale
+    cdef bint to_180
 
     def __cinit__(self):
         self.src_scale = 1
         self.dest_scale = 1
+        self.to_180 = False
 
-    cdef void init(self, projPJ src_proj, projPJ dest_proj):
-        self.src_proj = src_proj
-        self.dest_proj = dest_proj
+    cdef void init(self, src_crs, dest_crs) except *:
+        self.transformer = Transformer.from_crs(src_crs, dest_crs, always_xy=True)
+        self.to_180 = (
+            self.transformer.name == "noop" and
+            src_crs.__class__.__name__ in ("PlateCarree", "RotatedPole")
+        )
 
     cdef void set_line(self, const Point &start, const Point &end):
         self.start = start
         self.end = end
 
-    cdef Point interpolate(self, double t):
-        raise NotImplementedError
+    cdef Point project(self, const Point &src_xy) except *:
+        cdef Point dest_xy
+        cdef double src_xy_x = src_xy.x
+        if self.to_180 and src_xy_x > 180:
+            src_xy_x = (((src_xy_x + 180) % 360) - 180)
+
+        try:
+            xx, yy = self.transformer.transform(
+                src_xy_x * self.src_scale,
+                src_xy.y * self.src_scale,
+                errcheck=True
+            )
+        except ProjError as err:
+            msg = str(err).lower()
+            if (
+                "latitude" in msg or
+                "longitude" in msg or
+                "outside of projection domain" in msg or
+                "tolerance condition error" in msg
+            ):
+                xx = HUGE_VAL
+                yy = HUGE_VAL
+            else:
+                raise
+
+        dest_xy.x = xx * self.dest_scale
+        dest_xy.y = yy * self.dest_scale
+        return dest_xy
 
-    cdef Point project(self, const Point &point):
+    cdef Point interpolate(self, double t) except *:
```

Comment:
Why change the order of these? It makes reading the diff confusing.

Reply:
I moved it to the base class as you only need one project method with pyproj as it handles the differences internally.